WordPress/gutenberg
# Social Links Block: Add icon and background color options

## Description

Added options to the social links block allowing a user to specify the icon or background colors and have that apply to all the social links contained within it.

_Note: Saving the styles in the social link blocks will be done if this general approach is continued instead of switching to the useColors hook if possible._

https://github.com/WordPress/gutenberg/issues/21605

## How has this been tested?



Only manual testing.
WordPress: 5.5.1

#### Testing Instructions
1. Create a post adding social links block to it and adding urls for multiple platforms before saving.
2. Apply this PR
3. Reload the page, block should still be valid
4. Select the social links block
5. Select icon and background colors from the Inspector controls (include custom colors in testing)
6. Save post and reload editor. Colors should be maintained.
7. Switch style to logos only. The background color options should not be shown and when switching styles back the background color will have been cleared.

## Screenshots 
![SocialIcons-Colors](https://user-images.githubusercontent.com/60436221/93336348-fe023200-f86a-11ea-9d7e-2ce7fdda0313.gif)

## Types of changes




New feature. Enhancement.

## Next Steps
- [x] Add class names and styles to social link saving for front end display
- [x] ~~Switch to using `__experimentalUseColors` if its possible to achieve same editor functionality. (didn't work for me)~~
- [ ] Refine approach

## Checklist:
